test/functional/autocmd/cursormoved_spec.lua

```diff
@@ -31,4 +31,14 @@ describe('CursorMoved', function()
     eq({'aaa'}, funcs.nvim_buf_get_lines(eval('g:buf'), 0, -1, true))
     eq(0, eval('g:cursormoved'))
   end)
+
+  it("is not triggered by cursor movement prior to first CursorMoved instantiation", function()
+    source([[
+    let g:cursormoved = 0
+    " Move the cursor by inserting text
+    call nvim_feedkeys('aaa', 'i', v:true)
```

Comment:
`nvim_feedkeys()` does execute typeahead immediately and stays in Insert mode after typeahead is executed.
```suggestion
    normal aaa
```

Reply:
I'm not sure why this matters, since the test only needs to induce a shift in the cursor position (and expectedly fails on master like it should), but I changed it.

Reply:
Oh, actually even if this line is removed, this test still fails on `master`.

Reply:
You are right, I'm actually a bit confused about why this test would fail but the one above it would not?

Reply:
`w_last_cursormoved` is always updated when entering a new window:
<https://github.com/neovim/neovim/blob/95803f0e902278079e5876082465b1910c819947/src/nvim/window.c#L4594>

Reply:
Ah, thanks! makes sense. I simplified the test.